st2common/st2common/models/api/pack.py

```diff
@@ -48,67 +49,95 @@ class PackAPI(BaseAPI):
     model = PackDB
     schema = {
         'type': 'object',
+        'description': 'Content pack schema.',
         'properties': {
             'id': {
                 'type': 'string',
+                'description': 'Unique identifier for the pack.',
                 'default': None
             },
+            'name': {
+                'type': 'string',
+                'description': 'Display name of the pack. If the name only contains lowercase'
+                               'letters, digits and underscores, the "ref" field is not required.',
+                'required': True
+            },
             'ref': {
                 'type': 'string',
+                'description': 'Reference for the pack, used as an internal id.',
                 'default': None,
                 'pattern': PACK_REF_WHITELIST_REGEX
             },
-            "uid": {
-                "type": "string"
+            'uid': {
+                'type': 'string'
             },
-            'name': {
+            'description': {
                 'type': 'string',
+                'description': 'Brief description of the pack and the service it integrates with.',
                 'required': True
             },
-            'description': {
-                'type': 'string'
-            },
             'keywords': {
                 'type': 'array',
+                'description': 'Keywords describing the pack.',
                 'items': {'type': 'string'},
                 'default': []
             },
             'version': {
                 'type': 'string',
+                'description': 'Pack version. Must follow the semver format '
+                               '(for instance, "0.1.0").',
                 'pattern': PACK_VERSION_REGEX,
                 'required': True
             },
             'author': {
                 'type': 'string',
+                'description': 'Pack author or authors.',
                 'required': True
             },
             'email': {
                 'type': 'string',
-                'format': 'email',
-                'required': True
+                'description': 'E-mail of the pack author.',
+                'format': 'email'
             },
             'files': {
                 'type': 'array',
+                'description': 'A list of files inside the pack.',
+                'items': {'type': 'string'},
+                'default': []
+            },
+            'dependencies': {
+                'type': 'array',
+                'description': 'A list of other StackStorm packs this pack depends upon. '
+                               'The same format as in "st2 pack install" is used: '
+                               '"<name or full URL>[=<version or git ref>]".',
                 'items': {'type': 'string'},
                 'default': []
+            },
+            'system': {
```

Comment:
I would also simplify it and make it a string, I think there is little need to have an object since we have only one platform - StackStorm.

In Node.JS land it's a bit different since there are multiple executions environments using different engines / JITs. Having said that, I also very rarely seen people specify anything else than nodejs there :)